lib/battle/BattleHex.cpp

```diff
@@ -135,7 +136,7 @@ std::vector<BattleHex> BattleHex::neighbouringTiles() const
 {
 	std::vector<BattleHex> ret;
 	ret.reserve(6);
-	for(EDir dir = EDir(0); dir <= EDir(5); dir = EDir(dir+1))
+	for(EDir dir = static_cast<EDir>(0); dir <= static_cast<EDir>(5); dir = static_cast<EDir>(dir + 1))

```

Comment:
Is there an enum value for 5?

Reply:
Yes, all valid EDir's have valid values in enum, including 0 and 5. 
I am currently skipping cases like this during review.

We should probably recheck all our static_cast's. I expect *a lot* of places where static_cast is an indicator of poor code, like this one.
However I think this should go into a separate PR(s), for easier reviewing and minimal regressions